### ISSUE TITLE
Accessibility issue in Relational Databases Certification

### ISSUE THREAD

**Issue by arqamgrt** [freeCodeCamp/CodeAlly-CodeRoad-freeCodeCamp#55]
Hello Campers. Hope you’re doing well.
I’m Facing an issue in Relational Databases Certification (online).
There is no output spoken by screen readers in the terminal of vs code. I know, I can use it locally. I actually did it. But in the PostGreSQL Course, I was stuck on first step. I don’t know why, but it was failing the test. Same steps I tried online, and it was successful. But again, in online, there is no speech output for screen reader users of any command you type in the terminal.
I’m looking for a possibility to solve this accessibility issue. if not, then at least, in local, the bug should be fix. So, I’ll be able to complete the certification.
Looking forward to see your valuable responses on this issue.

**Comment by bbsmooth** [comment]
@arqamgrt I have not looked at this course yet but I will be happy to give it a try. Can you give us some more details about exactly where you are having problems. Since I have no experience with this course, please be as specific as possible. Also, what operating system, browser, and screen reader are you using?

**Comment by ShaunSHamilton** [comment]
Hello there,

Thank you for taking the time to open this. Unfortunately, we have no control over the VM serving the VSCode-style editor. I will ping @moT01 as we might need to transfer this issue to CodeAlly?

**Comment by arqamgrt** [comment]
> @arqamgrt I have not looked at this course yet but I will be happy to give it a try. Can you give us some more details about exactly where you are having problems. Since I have no experience with this course, please be as specific as possible. Also, what operating system, browser, and screen reader are you using?

Well, I'm using Windows 10 21H2. with chrome browser and NVDA screen reader.
if you talk about online mode, the only problem is that I can't hear the output of anything on terminal.

**Comment by moT01** [comment]
Thanks for reporting @arqamgrt, sorry for the delay here. I was sort of able to reproduce this with voiceover on mac. It reads output from the actual terminal, but not the terminal in VS Code - both locally and in CodeAlly. So when you use the terminal in your local VS Code, it reads the output for you? If so, do you know why I am not getting that? Maybe some extension or program you have, or maybe it's NVDA.

I will see if I can reproduce the issue you had with running these locally and get back to you.

**Comment by arqamgrt** [comment]
@moT01 Are you trying on Windows? or mac. If you want to try it on windows, then use NVDA for this.
but in local, I have a problem. in the PostGreSQL course, first step is not working in local. same step if working fine in code ally.

**Comment by moT01** [comment]
Right, I will look into that @arqamgrt. I am on a mac.

So, to clarify, do you get the terminal output read when using the terminal in VS Code anywhere? I'm trying to figure out if it's something with VS Code or CodeAlly.

**Comment by arqamgrt** [comment]
no no, I know this. vs code is completely fine here. the problem is only with codeAlly. in vs code, everything works fine including terminal.

**Comment by moT01** [comment]
I loaded the tutorial up locally for the first time in a while. I was unable to reproduce the getting stuck part of this issue. Sometimes it happens. There's some [troubleshooting steps to try here](https://forum.freecodecamp.org/t/running-the-relational-database-curriculum-in-your-browser/500231#issues-within-a-course-9) - otherwise, maybe just start over. The forum is the place to troubleshoot that, so if you are still stuck - I would make a post over there.

I am going to transfer this issue to the repo we have for tracking RDB issues that aren't fixable by fCC.

**Comment by moT01** [comment]
Not sure if this is a CodeAlly thing, or a VS Code thing, or other - but if you get a chance, maybe something to take a look at @AdamZaczek

**Comment by arqamgrt** [comment]
Look, When _ try to do it online, on CodeAlly. Accessibility issues are there. Terminal is not accessible. I can't read the output in the terminal.

**Comment by qkaweesi18** [comment]
I'm literally having the same issue and its driving me crazy, could have finished a lot of coursework by now. might look elsewhere for free certificates. what a HUGE dissapointment

**Comment by maryann-r** [comment]
Hello everyone.
OK. I also tried to follow this relational database tutorials, and I wanted to do them online, too. As I am facing the same issue, I thought I could bring a little light pasting here the message I've got when trying to enable accessibility on VS's CodeAlly version pressing alt+f1:
"Thank you for trying out VS Code's accessibility options.

Status:

- The editor is configured to use platform APIs to detect when a Screen Reader is attached, but the current runtime does not support this.

- Pressing Tab in the current editor will insert the tab character. Toggle this behavior by pressing Control+M.

Press Control+H now to open a browser window with more VS Code information related to Accessibility.

You can dismiss this tooltip and return to the editor by pressing Escape or Shift+Escape."
As it was mentioned before, it looks like we can write on the terminal, but not reading the terminal output, so we cannot check with the screen reader what's there. We don't get the same results when we execute it offline with local VS Code with Docker and all that.
Hope it is helpful.
